### ISSUE TITLE
Uncaught DataCloneError: Failed to execute 'postMessage' on 'Worker': An object could not be cloned

### ISSUE THREAD

**Issue by screendriver** [medialize/sass.js#39]
I am using sass.js in one of my Cordova projects running on an Android device. I use `sass.js/dist/sass.js` that uses a WebWorker. Unfortunately I get a `Uncaught DataCloneError: Failed to execute 'postMessage' on 'Worker': An object could not be cloned` error message.

Is there something I can do? On my desktop Browser it works as expected.

**Comment by rodneyrehm** [comment]
> I am using sass.js in one of my Cordova projects running on an Android device.

May I ask what you're using sass.js for specifically?

> Unfortunately I get a Uncaught `DataCloneError: Failed to execute 'postMessage' on 'Worker': An object could not be cloned error message.`

I think in the old days of WebWorkers we could only pass strings back and forth.

> Is there something I can do? On my desktop Browser it works as expected.

You could try using [Crosswalk](https://crosswalk-project.org/) to have a modern Chrome based WebView in your Cordova app.

Alternatively you could try replacing all the `.postMessage( … )` calls in [src/sass.js](https://github.com/medialize/sass.js/blob/master/src/sass.js) and [src/sass.worker.js](https://github.com/medialize/sass.js/blob/master/src/sass.worker.js) by `.postMessage(JSON.stringify( … ))` and make sure to `JSON.parse()` the event in [`_handleWorkerMessage()`](https://github.com/medialize/sass.js/blob/master/src/sass.js#L66) and [`onmessage()`](https://github.com/medialize/sass.js/blob/master/src/sass.worker.js#L28). Bonus points for first trying the unserialized object and only performing the serialization on failure. Here's how you [build sass.js yourself](https://github.com/medialize/sass.js#building-sassjs). PR welcome :)

**Comment by screendriver** [comment]
I wrote a [SystemJS](https://github.com/systemjs/systemjs) plugin called [plugin-sass](https://github.com/mobilexag/plugin-sass) because with it I can do a ES2015 `import 'mysassfile.scss!';` in my App and at runtime (or bundling time) my SASS files are compiled and injected right into the given html file.

Hmm I am using the latest WebView on my Android device / Emulator. It can't be newer because it uses the latest Chrome for Android runtime engine.

Something strange happened: I played with it a lot the whole day and now it's working :confused: I can't reproduce it. I don't know why. I close this issue at and reopen it when the error occurs again.
